app/src/main/java/com/example/spork/home/HomeFragment.java

```diff
@@ -1,66 +1,336 @@
+                                        .setInterval(10000)
+                                        .setFastestInterval(
+                                                1000)
+                                        .setNumUpdates(1);
+
+                                // Initialize location call back
+                                LocationCallback
+                                        locationCallback
+                                        = new LocationCallback() {
+                                    @Override
+                                    public void
+                                    onLocationResult(
+                                            LocationResult
+                                                    locationResult)
+                                    {
+                                        // Initialize
+                                        // location
+                                        Location location1
+                                                = locationResult
+                                                .getLastLocation();
+                                        // Set latitude
+                                        currentLat = location1.getLatitude();
+
+                                        // Set longitude
+                                        currentLng = location1.getLongitude();
+
+                                    }
+
+
+                                };
+
+                                // Request location updates
+                                client.requestLocationUpdates(
+                                        locationRequest,
+                                        locationCallback,
+                                        Looper.myLooper());
+                            }
+
+                            ParseGeoPoint currentLocation = new ParseGeoPoint(currentLat, currentLng);
+                            ParseUser.getCurrentUser().put("currentLocation", currentLocation);
+
+                            LatLng currentLatLng = new LatLng(currentLat, currentLng);
+                            Log.i(TAG, "currentLat: " + currentLat + " currentLng: " + currentLng);
+                            mMap.addMarker(new MarkerOptions().position(currentLatLng).title("Current Location").icon(BitmapDescriptorFactory.fromResource(R.drawable.ic_current_location)));
+                            mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(currentLatLng, zoom));
+
+                            StringBuilder sb = new StringBuilder("https://maps.googleapis.com/maps/api/place/nearbysearch/json");
+                            sb.append("?fields=name%2Cgeometry/location");
+                            sb.append("&location=" + currentLat + "%2C" + currentLng);
+                            sb.append("&radius=" + radius);
+                            sb.append("&type=restaurant");
+                            sb.append("&key=" + BuildConfig.MAPS_API_KEY);
```

Comment:
I'm not sure how often you'll be making this api call, but if you need to generate urls like this a lot, it might be helpful to create util file for generating these urls